crates/miden-lib/asm/miden/auth/multisig.masm

```diff
@@ -0,0 +1,282 @@
+        # => [OWNER_PUB_KEY, i-1, MSG]
+
+        # Check if signature exists for this signer.
+        # -----------------------------------------------------------------------------------------
+
+        movup.4 movdn.8 swapw dupw movdnw.2
+        # => [MSG, OWNER_PUB_KEY, MSG, i-1]
+
+        hmerge
+        # => [SIG_KEY, MSG, i-1]
+
+        adv.has_mapkey
+        # => [SIG_KEY, MSG, i-1]
+
+        adv_push.1
+        # => [has_signature, SIG_KEY, MSG, i-1]
+
+        # if h(msg, pk) => msg exists in AdviceMap check the signature
+        if.true
+            dropw dupw
+            # => [MSG, MSG, i-1]
+
+            # Verify the signature against the public key and message.
+            # -----------------------------------------------------------------------------------------
+
+            padw loc_loadw.TEMP_PK
+            # => [PK, MSG, MSG, i-1]
+
+            swapw
+            # => [MSG, PK, MSG, i-1]
+
+            # Emit the authentication request event that pushes a signature for the message to the advice stack.
+            emit.AUTH_REQUEST
+
+            swapw
+            # => [PK, MSG, MSG, i-1]
+
+            exec.rpo_falcon512::verify
+            # => [MSG, i-1]
+
+            loc_load.SUCCESSFUL_VERIFICATIONS
+            add.1
+            loc_store.SUCCESSFUL_VERIFICATIONS
+            # => [MSG, i-1]
+
+        else
+            dropw
+            # => [MSG, i-1]
+        end
+        # => [MSG, i-1]
+
+        movup.4
+        # => [i-1, MSG]
+
+        dup neq.0
+        # => [should_continue, i-1, MSG]
+    end
+    # => [i-1, MSG]
+
+    # Check number of successful signature verifications
+    # -----------------------------------------------------------------------------------------
+
+    drop
+    # => [MSG]
+
+    loc_load.SUCCESSFUL_VERIFICATIONS
+    # => [num_verifications, MSG]
+
+    loc_load.THRESHOLD
+    # => [threshold, num_verifications, MSG]
+
+    gte not
+    # => [is_unauthorized, MSG]
+
+    # If signatures are non-existent the tx will fail here.
+    if.true
+        emit.UNAUTHORIZED_EVENT
+        push.1 assertz.err="insufficient number of signatures"
+    end
+    # => [MSG]
+
+    dropw
+    # => []
+end
+
+#! Authenticate a transaction using the Falcon signature scheme with multi-signature support.
+#!
+#! This procedure implements multi-signature authentication by:
+#! 1. Computing the transaction summary message that needs to be signed
+#! 2. Verifying signatures from multiple required signers against their public keys
+#! 3. Ensuring the minimum threshold of valid signatures is met
+#! 4. Implementing replay protection by tracking executed transactions
+#!
+#! The procedure first increments the account nonce, then computes and verifies signatures for
+#! the following message (in memory order):
+#! [ACCOUNT_DELTA_COMMITMENT, INPUT_NOTES_COMMITMENT,
+#!  OUTPUT_NOTES_COMMITMENT, [0, 0, ref_block_num, final_nonce]]
+#!
+#! Inputs:
+#!   Operand stack: [SALT]
```

Comment:
I think the description of the `SALT` above in the signed message is inconsistent with how it is used in tests (which are the only reference currently). The `SALT` is expected to be passed as an auth argument and it would be good to describe why. Based on what was discussed in https://github.com/0xMiden/miden-base/issues/1534, we probably want to end up with a design that allows for multiple inflight multisig transactions. Not necessary to solve for this PR, but if we intend to ship this state of the multisig with the next release, we should state the assumptions for `SALT` and replace `[0, 0, ref_block_num, final_nonce]` with `SALT`.

For now, we could just say it must be a randomly generated value or something to that effect, but you've probably thought more about it already.